SpatialGDK/Source/SpatialGDK/Private/Interop/GlobalStateManager.cpp

```diff
@@ -537,13 +539,10 @@ void UGlobalStateManager::BecomeAuthoritativeOverAllActors()
 	for (TActorIterator<AActor> It(NetDriver->World); It; ++It)
 	{
 		AActor* Actor = *It;
-		if (Actor != nullptr && !Actor->IsPendingKill())
+		if (Actor != nullptr && !Actor->IsPendingKill() && USpatialWorker::CanHaveAuthority(Actor) && Actor->GetIsReplicated())
```

Comment:
Need to add additional check here to account for singletons being correctly hooked up on non-auth servers. i.e. should be:

`		if (Actor != nullptr && !Actor->IsPendingKill() && (Actor->GetClass()->HasAnySpatialClassFlags(ESpatialClassFlags::SPATIALCLASS_Singleton) || Actor->GetClass()->WorkerAssociation.Equals(WorkerType) || (Actor->GetClass()->WorkerAssociation.IsEmpty() && WorkerType.Equals(SpatialConstants::ServerWorkerType))))`